### PR TITLE
Fixing perf issue with getitem by Boolean mask

### PR DESCRIPTION
**Pull request by kozlov-alexey**

### CODE REVIEW

File: sdc/datatypes/hpat_pandas_dataframe_functions.py [hunk 1]
```diff
@@ -1454,13 +1454,16 @@ def df_length_expr(self):
...
-def df_index_expr(self, length_expr=None):
+def df_index_expr(self, length_expr=None, as_range=False):
     """Generate expression to get or create index of DF"""
     if isinstance(self.index, types.NoneType):
         if length_expr is None:
             length_expr = df_length_expr(self)
 
-        return f'numpy.arange({length_expr})'
+        if as_range:
+            return f'range(0, {length_expr})'
```
Comment (densmirn) [inline sdc/datatypes/hpat_pandas_dataframe_functions.py]: Why not
```suggestion
            return f'range({length_expr})'
```
?

File: sdc/datatypes/hpat_pandas_dataframe_functions.py [hunk 2]
```diff
@@ -1532,7 +1536,7 @@ def df_getitem_bool_series_idx_main_codelines(self, idx):
                       f'  self_index = self.index',
                       f'  idx_reindexed = sdc_reindex_series(idx._data, idx.index, idx._name, self_index)',
                       f'  res_index = getitem_by_mask(self_index, idx_reindexed._data)',
-                      f'  selected_pos = getitem_by_mask(numpy.arange(length), idx_reindexed._data)']
+                      f'  selected_pos = getitem_by_mask(range(0, length), idx_reindexed._data)']
```
Comment (densmirn) [inline sdc/datatypes/hpat_pandas_dataframe_functions.py]: Looks like it is equal to:
```suggestion
                      f'  selected_pos = getitem_by_mask(range(length), idx_reindexed._data)']
```

File: sdc/datatypes/hpat_pandas_series_functions.py [hunk 3]
```diff
@@ -430,7 +430,7 @@ def hpat_pandas_series_getitem_idx_bool_indexer_impl(self, idx):
 
                 return pandas.Series(
                     data=numpy_like.getitem_by_mask(self._data, idx._data),
-                    index=numpy_like.getitem_by_mask(self.index, idx._data),
+                    index=numpy_like.getitem_by_mask(range(0, len(self)), idx._data),
```
Comment (densmirn) [inline sdc/datatypes/hpat_pandas_series_functions.py]:
```suggestion
                    index=numpy_like.getitem_by_mask(range(len(self)), idx._data),
```

**Comment by kozlov-alexey** [comment]
> any performance data?

@AlexanderKalistratov TC builds are failing now, so I re-measured manually on nnlmlp01 (the data size was increased to 5 * 10 ** 7 to increase exec times):
With fix:
                                                              median       min       max   compile    boxing
name                              nthreads type   size
DataFrame.getitem_idx_bool_array  1        Python 50000000  0.729936  0.728974  0.734385       NaN       NaN
                                           SDC    50000000  0.783687  0.780611  0.789920  0.454833  2.437608
                                  2        SDC    50000000  0.420206  0.418938  0.420460  0.466937  2.445182
                                  4        SDC    50000000  0.229900  0.228938  0.245967  0.471535  2.501849
                                  8        SDC    50000000  0.140959  0.140459  0.141460  0.569357  2.520889
                                  16       SDC    50000000  0.114258  0.109348  0.155166  0.488567  2.548175
                                  28       SDC    50000000  0.093318  0.092599  0.094481  0.527987  2.517458
                                  56       SDC    50000000  0.084853  0.082968  0.086843  0.469488  2.520272
DataFrame.getitem_idx_bool_series 1        Python 50000000  0.727004  0.726642  0.730250       NaN       NaN
                                           SDC    50000000  0.667441  0.660132  0.678571  0.464186  2.483425
                                  2        SDC    50000000  0.344520  0.343004  0.345930  0.481691  2.452096
                                  4        SDC    50000000  0.185001  0.183356  0.196492  0.606666  2.491350
                                  8        SDC    50000000  0.111713  0.105505  0.126037  0.421492  2.495467
                                  16       SDC    50000000  0.075595  0.065445  0.083412  0.494176  2.525917
                                  28       SDC    50000000  0.065051  0.064746  0.073442  0.446053  2.467280
                                  56       SDC    50000000  0.066710  0.065066  0.067768  0.430038  2.518629

On master (without fix):
                                                              median       min       max   compile    boxing
name                              nthreads type   size
DataFrame.getitem_idx_bool_array  1        Python 50000000  0.709037  0.705619  0.712697       NaN       NaN
                                           SDC    50000000  1.295295  1.249202  1.332165  0.573321  2.476892
                                  2        SDC    50000000  0.734991  0.730225  0.774579  0.533662  2.522519
                                  4        SDC    50000000  0.430704  0.404743  0.436254  0.520294  2.499454
                                  8        SDC    50000000  0.271650  0.268377  0.279392  0.593541  2.522613
                                  16       SDC    50000000  0.193998  0.190620  0.199090  0.607101  2.523254
                                  28       SDC    50000000  0.183823  0.183236  0.184575  0.449276  2.489915
                                  56       SDC    50000000  0.182439  0.173317  0.190032  0.655567  2.558420
DataFrame.getitem_idx_bool_series 1        Python 50000000  0.722556  0.719484  0.724653       NaN       NaN
                                           SDC    50000000  0.887583  0.883327  0.891517  0.519036  2.510130
                                  2        SDC    50000000  0.515719  0.514620  0.519068  0.477245  2.470925
                                  4        SDC    50000000  0.281323  0.280721  0.289484  0.478673  2.472751
                                  8        SDC    50000000  0.179992  0.178943  0.183958  0.452823  2.467900
                                  16       SDC    50000000  0.130896  0.127235  0.139556  0.462505  2.516520
                                  28       SDC    50000000  0.098136  0.096969  0.108245  0.496178  2.503406
                                  56       SDC    50000000  0.096119  0.094569  0.100459  0.442157  2.560549

**Pull request merged**